File: src/src/org/github/olloginov/ideataskbrowser/util/TaskHelper.java
package org.github.olloginov.ideataskbrowser.util;

import com.intellij.tasks.Task;

import java.util.Date;

public abstract class TaskHelper {
    public static Date getChangeDate(Task task) {
        return max(task.getCreated(), task.getUpdated());
    }

    public static Date max(Date a, Date b) {
        if (a == null || b == null) {
            return a == null ? b : a;
        }
        return a.after(b) ? a : b;
    }
}

File: src/main/kotlin/org/github/olloginov/ideataskbrowser/util/TaskHelper.kt
package org.github.olloginov.ideataskbrowser.util

import com.intellij.tasks.Task
import com.intellij.tasks.gitlab.GitlabTask
import com.intellij.tasks.gitlab.model.GitlabIssue
import java.lang.reflect.InaccessibleObjectException
import java.util.Date

object TaskHelper {
    fun getChangeDate(task: Task): Date? {
        return max(task.created, task.updated)
    }

    private fun max(a: Date?, b: Date?): Date? {
        if (a == null || b == null) {
            return a ?: b
        }
        return if (a.after(b)) a else b
    }

    fun getPresentationTitle(task: Task): String {
        return if (task.isIssue) {
            task.presentableId + ": " + task.summary
        } else {
            task.summary
        }
    }

    fun getGitlabDescription(task: GitlabTask): String? {
        return try {
            val field = task.javaClass.getDeclaredField("myIssue")
            field.isAccessible = true
            val issue = field.get(task) as GitlabIssue
            issue.description
        } catch (e: NoSuchFieldException) {
            null
        } catch (e: InaccessibleObjectException) {
            null
        } catch (e: SecurityException) {
            null
        }
    }
}
